game/cards/dm06/hedrian.go

```diff
@@ -0,0 +1,75 @@
+package dm06
+
+import (
+	"duel-masters/game/civ"
+	"duel-masters/game/cnd"
+	"duel-masters/game/family"
+	"duel-masters/game/fx"
+	"duel-masters/game/match"
+	"fmt"
+)
+
+func TankMutant(c *match.Card) {
+
+	c.Name = "Tank Mutant"
+	c.Power = 6000
+	c.Civ = civ.Darkness
+	c.Family = []string{family.Hedrian}
+	c.ManaCost = 9
+	c.ManaRequirement = []string{civ.Darkness}
+	c.TapAbility = true
+
+	c.Use(fx.Creature, fx.When(fx.TapAbility, func(card *match.Card, ctx *match.Context) {
+		fx.Select(
+			ctx.Match.Opponent(card.Player),
+			ctx.Match,
+			ctx.Match.Opponent(card.Player),
+			match.BATTLEZONE,
+			"Tank Mutant: Select 1 creature from your battlezone that will be sent to your graveyard",
+			1,
+			1,
+			false,
+		).Map(func(creature *match.Card) {
+			ctx.Match.Destroy(creature, card, match.DestroyedByMiscAbility)
+		})
+
+		card.Tapped = true
+	}))
+
+}
+
+func BazookaMutant(c *match.Card) {
+
+	c.Name = "Bazooka Mutant"
+	c.Power = 8000
+	c.Civ = civ.Darkness
+	c.Family = []string{family.Hedrian}
+	c.ManaCost = 4
+	c.ManaRequirement = []string{civ.Darkness}
+
+	c.Use(fx.Creature, fx.CantAttackPlayers, func(card *match.Card, ctx *match.Context) {
+
+		if card.Zone != match.BATTLEZONE {
+			return
+		}
+
+		if event, ok := ctx.Event.(*match.AttackCreature); ok {
+
+			if !ok || event.CardID != card.ID {
+				return
+			}
+
+			for _, creature := range event.AttackableCreatures {
+
+				if !creature.HasCondition(cnd.Blocker) {
+
+					ctx.Match.WarnPlayer(card.Player, fmt.Sprintf("%s can't attack creatures other than blockers", card.Name))
+					ctx.InterruptFlow()
+					return
+				}
+
+			}
+
+		}
```

Comment:
If you meant, that when there are few blockers and one non-blocker, than it works correctly.
Same if there is only one blocker and few non-blockers,
It just can attack cards that are blockers && can be attackable (ex. tapped)
So no attacking non-blockers and no attacking untapped, if that also what you meant

Reply:
When looping over `event.AttackableCreatures` the function will `return` and cancel the event if any of the attackable cards are not a blocker. So in a scenario where the opponent has 1 tapped non-blocker and 1 tapped blocker it should be possible to attack the blocker but since the code checks if *all* the attackable cards have blocker or not then it won't work.

I think to solve this it needs to overwrite the `event.AttackableCreatures` to only include cards that have blocker, similar to how cards that can only attack a certain civ does it (For example `Photocide, Lord of the Wastes`)